sdk/storage/Azure.Storage.Common/src/Shared/StorageRequestFailedDetailsParser.cs

```diff
@@ -46,7 +47,15 @@ public override bool TryParse(Response response, out ResponseError? error, out I
                             }
                         }
 
-                        error = new ResponseError(errorCode, message);
+                        if (headerName != null && headerName == Constants.HeaderNames.Version)
```

Comment:
Should we maybe also check for the InvalidHeaderValue error code just to be sure?

Reply:
Fixed.